src/plugins/blocksmith/components/instagram-media-block-preview/index.jsx

```diff
@@ -1,17 +1,90 @@
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import Message from '@gdbots/pbj/Message';
-import InstagramEmbed from 'react-instagram-embed';
 
-const InstagramMediaBlockPreview = ({ block }) => (
-  <InstagramEmbed
-    url={`https://instagr.am/p/${block.get('id')}/`}
-    hideCaption={block.get('hidecaption') || false}
-  />
-);
+export default class InstagramPostBlockPreview extends Component {
+  static propTypes = {
+    block: PropTypes.instanceOf(Message).isRequired,
+  };
 
-InstagramMediaBlockPreview.propTypes = {
-  block: PropTypes.instanceOf(Message).isRequired,
-};
+  constructor(props) {
+    super(props);
+    this.embedParentRef = React.createRef();
+    this.clean = this.clean.bind(this);
+    this.embed = this.embed.bind(this);
+    this.state = {
+      instagramID: '',
+      hideCaption: false,
+    }
+  }
 
-export default InstagramMediaBlockPreview;
+  componentDidMount() {
+    const { block } = this.props;
+    this.setState({
+      instagramID: block.get('id'),
+      hideCaption: block.get('hidecaption'),
+    }, () => {
+      const script = document.createElement('script');
+      script.async = true;
+      script.src = '//www.instagram.com/embed.js';
+      script.setAttribute('charset', 'utf-8');
+      script.onload = () => {
+        instgrm.Embeds.process();
+      }
+      this.embedParentRef.current.appendChild(script);
+      this.embed();
+    })
+  }
+
+  componentDidUpdate() {
```

Comment:
put this in a separate function and then call that func in this lifecycle method. ref how imgur does it